### ISSUE TITLE
find/replace overlay: return to last cursor position when emptying the search

### ISSUE THREAD

**Issue by Wittmaxi** [eclipse-platform/eclipse.platform.ui#1944]
When typing into the search field, the cursor is moved around the editor, displaying the results of an incremental search.
When completely emptying the "search"-field, the cursor should be reset to what it was before entering the find/field.

Here is a video showcasing my frustration with this behavior:
![6](https://github.com/eclipse-platform/eclipse.platform.ui/assets/16443184/ad70cef9-429f-40cb-9769-456bc8ec30d8)

By "accidentally" pressing "0" while inside of the search field, I am suddenly on the complete other end of the file.

---
VSCode handles this case exactly like the current Overlay does. I could not find a plausible reason why, I believe that this is a genuine pain point where as you want to be able return to the position where you started searching before typing anything into the search bar.

**Comment by deepika-u** [comment]
Hi @Wittmaxi,
I am trying to fix this issue but need some inputs here.

updateIncrementalSearch() is the basic call invoked when the find string is updated. Via this i have made some changes to incrementalBaseLocation updation/tracking/storing and able to retain the original position in the editor where the cursor was pointing before the search(find/replace overlay was open). But i need to pass on this result to some call/logic which i am struggling to figure out. Basically i need some inputs on which exact function call or logic is responsible for **highlighting/using the last search result** in editor once the overlay is closed?  Can you share your thoughts please.

**Comment by deepika-u** [comment]
@Wittmaxi,
Do you have any inputs for me please?

**Comment by Wittmaxi** [comment]
@deepika-u yes, thank you for taking interest in this issue. I have stopped my official work at Vector Informatik to focus on my academic studies, hence why I am less involved than a few weeks ago.
Nonetheless, I am still always happy to contribute to Eclipse, in particular I am happy to help other people get on board this feature :)

The FindReplaceTargetExtension3 ( I believe it was this one ) has a method to set a particular selection in the target (here: the editor). I would not manipulate the incremental search.
What I would do is:

1) if a letter was entered into the input bar and the input bar was empty before, store the current selection (can have width 0)
2) once the input bar is empty again, use the FindReplaceTargetExtension3 to reset the previous selection

If 2) doesn't work (I don't remember exactly if this was possible), what you should do is that you store the position and the string of the selection. instead of 2), you do 2b): set the "incremental search location" to the stored position and perform a search for the string that was stored.

2b) is a hack, 2) would be preferrable

**Comment by deepika-u** [comment]
Hey! that was cool. All the best. I'll check on your pointers and get back. Thanks
